### PR TITLE
RavenDB-7650 implementation

### PR DESCRIPTION
**Pull request by myarichuk**
* add query timings to response headers
* if query latency was more than (configurable) threshold, create alert

### CODE REVIEW

File: src/Raven.Server/Config/Categories/CoreConfiguration.cs [hunk 1]
```diff
@@ -46,6 +46,12 @@ public class CoreConfiguration : ConfigurationCategory
         [ConfigurationEntry("ThrowIfAnyIndexCannotBeOpened", ConfigurationEntryScope.ServerWideOrPerDatabase)]
         public bool ThrowIfAnyIndexCannotBeOpened { get; set; }
 
+        [Description("Indicates request latency threshold before the server would issue an alert")]
+        [ConfigurationEntry("TooLongRequestThreshold", ConfigurationEntryScope.ServerWideOnly)]
+        [DefaultValue(2)]
```
Comment (ayende) [inline src/Raven.Server/Config/Categories/CoreConfiguration.cs:51]: Let's set the default to 5, 2 seems too low.

File: src/Raven.Server/Web/RequestHandler.cs [hunk 2]
```diff
@@ -353,12 +358,47 @@ private static void InvalidCountOfValues(string name)
...
+                            _context.RavenServer.ServerStore.NotificationCenter.Add(alert);
+                        }
+                    }
+                }
+                catch (Exception e)
+                {
+                    //precaution - should never arrive here
+                    if (_logger.IsInfoEnabled)
+                    {
+                        _logger.Info($"Failed to write request time in response headers. This is not supposed to happen and is probably a bug. The request path was: {_context.HttpContext.Request.Path}",e);
```
Comment (ayende) [inline src/Raven.Server/Web/RequestHandler.cs:398]: This isn't what is going on here at all. This is actually the error if we can't raise the alert.

File: src/Raven.Server/Web/RequestHandler.cs [hunk 3]
```diff
@@ -353,12 +358,47 @@ private static void InvalidCountOfValues(string name)
...
+            }, _context.HttpContext);
+
+            return new DisposableAction(() =>
+            {
+                try
+                {
+                    if (alertThresholdExceeded)
+                    {
+                        var threshold = _context.RavenServer.Configuration.Core.TooLongRequestThreshold.AsTimeSpan;
+                        if (threshold.TotalMilliseconds >= sw.ElapsedMilliseconds)
```
Comment (ayende) [inline src/Raven.Server/Web/RequestHandler.cs:378]: This is not a good idea in this manner.
For example, we routinely have requests that last 15 seconds, any request to create an index that is waiting for it, for example.

I think that the default should be 30 seconds.

File: src/Raven.Server/Config/Categories/CoreConfiguration.cs (hunk 1 above)
Comment (ppekrol) [reply src/Raven.Server/Config/Categories/CoreConfiguration.cs:51]: Should be moved to PerformanceHintsConfiguration? Key should be 'PerformanceHints.TooLongRequestThresholdInSec' for consistency.

File: src/Raven.Server/Web/RequestHandler.cs [hunk 4]
```diff
@@ -353,12 +358,47 @@ private static void InvalidCountOfValues(string name)
...
+                        {
+
+                            var alert = AlertRaised.Create(
```
Comment (ppekrol) [inline src/Raven.Server/Web/RequestHandler.cs:381]: In my opinion it should be a PerformanceHint and it should aggregate itself like we do with Paging performance hints.

File: src/Raven.Server/Config/Categories/CoreConfiguration.cs (hunk 1 above)
Comment (ayende) [reply src/Raven.Server/Config/Categories/CoreConfiguration.cs:51]: Yes, good idea

**Pull request merged**